src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyoutViewModel.cs

```diff
@@ -206,20 +222,64 @@ public bool IsRenameOverloadsEditable
         public bool IsRenameOverloadsVisible
             => Session.HasRenameOverloads;
 
+        public bool AllowUserInput
+            => !Session.IsCommitInProgress;
+
+        private Visibility _visibility;
+        public Visibility Visibility
+        {
+            get => _visibility;
+            set
+            {
+                if (value != _visibility)
+                {
+                    Set(ref _visibility, value);
+                    NotifyPropertyChanged();
+                }
+            }
+        }
+
         public TextSpan StartingSelection { get; }
 
         public bool Submit()
         {
+            using var token = _asyncListener.BeginAsyncOperation(nameof(Submit));
             if (StatusSeverity == Severity.Error)
             {
                 return false;
             }
 
             SmartRenameViewModel?.Commit(IdentifierText);
-            Session.Commit();
+            _ = CommitAsync().ReportNonFatalErrorAsync();
             return true;
         }
 
+        private async Task CommitAsync()
+        {
+            try
+            {
+                await Session.CommitAsync(previewChanges: false).ReportNonFatalErrorAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex) when (FatalError.ReportAndCatch(ex, ErrorSeverity.Critical))
```

Comment:
why do we need this cache if we already hve ReportNonFatalErrorAsync above?

Reply:
need to at least doc to explain.

Reply:
: ) Removed. It will be reported via `ReportNonFatalErrorAsync`